### PR TITLE
Create plugin that sorts a referenced dictionary #733

### PR DESCRIPTION
**Pull request by sameersecond**
Create plugin that sorts a referenced dictionary #733

Added the plugin and the test.
----

**Please** following this checklist to help us incorporate your contribution quickly and easily:

 - [x] Give the same name to the pull request as the name of ticket 
 - [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [x] Test your code throughly before PR.

To make clear that you license your contribution under the MIT you have to acknowledge this by using the following check-box.

 - [x] I hereby declare this contribution to be licenced under the [MIT license](https://opensource.org/licenses/MIT)

PRs without ticked MIT license will not be accepted - Sorry.

### CODE REVIEW

File: tracardi/process_engine/action/v1/sort_dictionary.py [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+    def if_order_is_empty(cls, value):
+        if value == "":
+            raise ValueError("Direction cannot be empty")
+        return value
+
+    @validator("direction")
+    def if_order_has_valid_value(cls, value):
+        if value == "asc" or value == "desc":
+            return value
+        raise ValueError("Direction has invalid value")
```
Comment (atompie) [inline tracardi/process_engine/action/v1/sort_dictionary.py:24]: Could you add values that are possible (like: Possible values 'asc', 'desc') in the error message.

File: tracardi/process_engine/action/v1/sort_dictionary.py [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+
+    async def run(self, payload: dict, in_edge=None):
+        dot = self._get_dot_accessor(payload)
+        dictionary_to_sort = dot[self.config.data]
+
+        sorted_tuple_list = []
+        is_desc = self.config.direction == "desc"
+        for i in sorted(dictionary_to_sort, key=dictionary_to_sort.get, reverse=is_desc):
+            sorted_tuple_list.append(tuple([i, dictionary_to_sort[i]]))
+        return Result(port="payload", value=sorted_tuple_list)
```
Comment (atompie) [inline tracardi/process_engine/action/v1/sort_dictionary.py:52]:
```
return Result(port="payload", value={"result":sorted_tuple_list})
```

Result value should be a dict.

**Review by atompie — CHANGES_REQUESTED** [review]
You nailed it :) Only some manor changes requested and I am ready to merge.

File: test/unit/plugins/test_plugin_sort_dictionary.py [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+def test_sort_dictionary_plugin():
+    properties = {"data": {"first": 1, "second": 2, "fifth": 5, "third": 3}}
+    init = dict(
+        data='event@properties.data',
+        direction='asc',
+    )
+
+    result = run_plugin(SortedDictAction, init, properties)
+    assert result.output.value == {[["first", 1], ["second", 2], ["third", 3], ["fifth", 5]]}
```
Comment (atompie) [inline test/unit/plugins/test_plugin_sort_dictionary.py:13]: I see the changes in the plugin regarding the result. I believe that  after the changes this test will fail.

**Review by atompie — CHANGES_REQUESTED** [review]
Sorry I spotted one more thing.

File: test/unit/plugins/test_plugin_sort_dictionary.py (hunk 3 above)
Comment (sameersecond) [reply test/unit/plugins/test_plugin_sort_dictionary.py:13]: Ahh. You're right. Thanks for spotting that

**Comment by atompie** [comment]
Ok I will correct it.

**Review by atompie — APPROVED** [review]
Everything seems ok. Merging.

**Pull request merged**